### PR TITLE
ci: permit changes to doc/cloud/*

### PR DESCRIPTION
**Pull request by michaellzc**
In https://github.com/sourcegraph/sourcegraph/pull/59114, we started blocking all changes to docs v1.

After discussing with @MaedahBatool @eshamow, we decided to permit changes to Cloud docs until docs v1 can be sunset to avoid showing outdated content to Cloud customers.

## Test plan

CI should pass

### CODE REVIEW

**Comment by michaellzc** [comment]
Current dependencies on/for this PR:
* `main`
  * **PR #59189** <a href="https://app.graphite.dev/github/pr/sourcegraph/sourcegraph/59189?utm_source=stack-comment-icon" target="_blank">Graphite</a>  👈

This [stack of pull requests](https://stacking.dev/?utm_source=stack-comment) is managed by [Graphite](https://app.graphite.dev/github/pr/sourcegraph/sourcegraph/59189?utm_source=stack-comment).

File: doc/cloud/index.md [hunk 1]
```diff
@@ -110,7 +110,7 @@ If you have specific requirements for the region, please reach out to your Sourc
 
 More details about the locations and data storage can be found in [our handbook](https://handbook.sourcegraph.com/departments/cloud/technical-docs/multi-region/)
 
-### Private Connectivity
+### Private Connectivit (test remote later)
```
Comment (eseliger) [inline doc/cloud/index.md:113]: is this change intentional?

**Review by eseliger — APPROVED** [review]

File: doc/cloud/index.md (hunk 1 above)
Comment (michaellzc) [reply doc/cloud/index.md:113]: yeah going to revert later after pasing the build as test plan

**Review by jamesmcnamara — APPROVED** [review]

**Review by jamesmcnamara — CHANGES_REQUESTED** [review]
Sorry my github UI is currently not loading very effectively so I can't leave this inline.

Currently the doubly-nested continue is causing the `continue` to have no effect. I think you'll need to wrap it in a function i.e.

```go
func isDiffAllowed(filename string) bool {
  for _, allowed := range allowDocChangesPrefixAllowlist {
    if strings.HasPrefix(allowed, filename) {
      return true
    }
  }
  return false
}
```

And then later:
```go
for filename := range diff {
  if !isDiffAllowed(filename) {
    diffset[filename] = struct{}{}
  }
}
```

**Review by jamesmcnamara — APPROVED** [review]

File: doc/cloud/index.md (hunk 1 above)
Comment (michaellzc) [inline doc/cloud/index.md:113]:
```suggestion
### Private Connectivity
```

**Pull request merged**